wallet/wallet_test.go

```diff
@@ -0,0 +1,133 @@
+	return nil, nil
+}
+
+var cs = consensus.State{
+	Index: consensus.ChainIndex{
+		Height: 1,
+		ID:     consensus.BlockID{},
+	},
+}
+
+// TestWalletRedistribute is a small unit test that covers the functionality of
+// the 'Redistribute' method on the wallet.
+func TestWalletRedistribute(t *testing.T) {
+	oneSC := types.SiacoinPrecision
+
+	// create a wallet with one output
+	priv := consensus.GeneratePrivateKey()
+	pub := priv.PublicKey()
+	utxo := wallet.SiacoinElement{
+		types.SiacoinOutput{
+			Value:      oneSC.Mul64(20),
+			UnlockHash: wallet.StandardAddress(pub),
+		},
+		randomOutputID(),
+		0,
+	}
+	s := &mockStore{utxos: []wallet.SiacoinElement{utxo}}
+	w := wallet.NewSingleAddressWallet(priv, s)
+
+	numOutputsWithValue := func(v types.Currency) (c uint64) {
+		utxos, _ := w.UnspentOutputs()
+		for _, utxo := range utxos {
+			if utxo.Value.Equals(v) {
+				c++
+			}
+		}
+		return
+	}
+
+	applyTxn := func(txn types.Transaction) {
+		for _, input := range txn.SiacoinInputs {
+			for i, utxo := range s.utxos {
+				if input.ParentID == types.SiacoinOutputID(utxo.ID) {
+					s.utxos[i] = s.utxos[len(s.utxos)-1]
+					s.utxos = s.utxos[:len(s.utxos)-1]
+				}
+			}
+		}
+		for _, output := range txn.SiacoinOutputs {
+			s.utxos = append(s.utxos, wallet.SiacoinElement{output, randomOutputID(), 0})
+		}
+	}
+
+	// assert number of outputs
+	if utxos, err := w.UnspentOutputs(); err != nil {
+		t.Fatal(err)
+	} else if len(utxos) != 1 {
+		t.Fatalf("unexpected number of outputs, %v != 1", len(utxos))
+	}
+
+	// split into 3 outputs of 6SC each
+	amount := oneSC.Mul64(6)
+	if txn, _, err := w.Redistribute(cs, 3, amount, types.NewCurrency64(1), nil); err != nil {
+		t.Fatal(err)
+	} else {
+		applyTxn(txn)
+	}
+
+	// assert number of outputs
+	if utxos, err := w.UnspentOutputs(); err != nil {
+		t.Fatal(err)
+	} else if len(s.utxos) != 4 {
+		t.Fatalf("unexpected number of outputs, %v != 4", len(utxos))
+	}
+
+	// assert number of outputs that hold 6SC
+	if cnt := numOutputsWithValue(amount); cnt != 3 {
+		t.Fatalf("unexpected number of 6SC outputs, %v != 3", cnt)
+	}
+
+	// split into 3 outputs of 7SC each, expect this to fail
+	_, _, err := w.Redistribute(cs, 3, oneSC.Mul64(7), types.NewCurrency64(1), nil)
+	if err == nil || !strings.Contains(err.Error(), "insufficient balance") {
+		t.Fatalf("unexpected err: '%v'", err)
+	}
+
+	// split into 2 outputs of 9SC
+	amount = oneSC.Mul64(9)
+	if txn, _, err := w.Redistribute(cs, 2, oneSC.Mul64(9), types.NewCurrency64(1), nil); err != nil {
+		t.Fatal(err)
+	} else {
+		applyTxn(txn)
+	}
+
+	// assert number of outputs
+	if utxos, err := w.UnspentOutputs(); err != nil {
+		t.Fatal(err)
+	} else if len(s.utxos) != 3 {
+		t.Fatalf("unexpected number of outputs, %v != 3", len(utxos))
+	}
+
+	// assert number of outputs that hold 9SC
+	if cnt := numOutputsWithValue(amount); cnt != 2 {
+		t.Fatalf("unexpected number of 9SC outputs, %v != 2", cnt)
+	}
+}
+
+func randomOutputID() (t types.OutputID) {
```

Comment:
Lately I've stopped doing this in tests. "random" makes it seem like it's fuzzing somehow, but really we just want these values to be *distinct*.

For a small test, I would just write `OutputID{1}`, `OutputID{2}`, etc. For a larger test, maybe a helper function like:

```go
var oidN uint32
outputID := func() (o types.OutputID) {
    oidN++
    binary.LittleEndian.PutUint32(o[:], oidN)
    return o
}
```

Eliminating the randomness also ensures that the test is completely deterministic.